src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/sensor/DetektorConfiguration.kt

```diff
@@ -16,55 +16,76 @@ import io.gitlab.arturbosch.detekt.sonar.foundation.NoAutoCorrectConfig
 
-	val pathFiltersString = settings.getString(PATH_FILTERS_KEY) ?: PATH_FILTERS_DEFAULTS
-	val filters = pathFiltersString.split(SEPARATOR_SEMICOLON, SEPARATOR_COMMA).map { PathFilter(it) }
+	val config: Config
+	val settings: Settings
+	val baseDir: File
 
-	val config = chooseConfig(baseDir, settings)
-	val processingSettings = ProcessingSettings(
-			baseDir.toPath(), NoAutoCorrectConfig(config), filters)
+	init {
+		val fileSystem = context.fileSystem()
+		baseDir = fileSystem.baseDir()
+		settings = context.settings()
+		config = chooseConfig(baseDir, settings)
+	}
 
-	return DetektFacade.instance(processingSettings)
-}
+	/**
+	 * Provides an instance of [Detektor] to be used for whole project analysis.
+	 */
+	fun configureDetektor(): Detektor {
+		val pathFiltersString = settings.getString(PATH_FILTERS_KEY) ?: PATH_FILTERS_DEFAULTS
+		val filters = pathFiltersString.split(SEPARATOR_SEMICOLON, SEPARATOR_COMMA).map { PathFilter(it) }
+		val processingSettings = ProcessingSettings(baseDir.toPath(), NoAutoCorrectConfig(config), filters)
 
-private fun chooseConfig(baseDir: File, settings: Settings): Config {
-	val configPath = settings.getString(CONFIG_PATH_KEY)
-	val externalConfigPath = configPath?.let {
-		LOG.info("Registered config path: $it")
-		val configFile = File(it)
-		if (!configFile.isAbsolute) { // TODO find out how to resolve always to root path, not module path
-			val resolved = baseDir.resolve(configPath)
-			LOG.info("Relative path detected. Resolving to project dir: $resolved")
-			resolved
-		} else configFile
+		return DetektFacade.instance(processingSettings)
 	}
 
-	val internalConfigResource = settings.getString(CONFIG_RESOURCE_KEY)
-			?.let { if (it.isNullOrBlank()) null else it }
+	/**
+	 * Provides an instance of [Detektor] for a specific [inputFile]
+	 */
+	fun configureDetektor(inputFile: InputFile): Detektor {
+		val processingSettings = ProcessingSettings(inputFile.path(), NoAutoCorrectConfig(config), emptyList())
 
-	val possibleParseArguments = Args().apply {
-		config = externalConfigPath?.path
-		configResource = internalConfigResource
+		return DetektFacade.instance(processingSettings)
 	}
 
-	val bestConfigMatch = possibleParseArguments.loadConfiguration()
-	val bestConfigMatchOrDefault = bestConfigMatch.let {
-		if (it == Config.empty) {
-			LOG.info("No detekt yaml configuration file found, using the default configuration.")
-			DEFAULT_YAML_CONFIG
-		} else it
+	private fun chooseConfig(baseDir: File, settings: Settings): Config {
+		val configPath = settings.getString(CONFIG_PATH_KEY)
+		val externalConfigPath = configPath?.let {
+			LOG.info("Registered config path: $it")
+			val configFile = File(it)
+			if (!configFile.isAbsolute) { // TODO find out how to resolve always to root path, not module path
+				val resolved = baseDir.resolve(configPath)
+				LOG.info("Relative path detected. Resolving to project dir: $resolved")
+				resolved
+			} else configFile
```

Comment:
That is something about code agreements. I tend to have curly braces for else (even one liner) when `if` part has curly braces already

Reply:
This is the part I'm not sure yet as having a single var reference inside an else block looks quite stupid to me. If the expression is complex or just has a `return` it looks better. But you are right, using curly braces should be preferred.

Reply:
Just try to write with and without and read it after some time. Without case is much harder to parse where else starts and what is inside